Kitodo-LongTermPreservationValidation/src/main/java/org/kitodo/longtermpreservationvalidation/jhove/KitodoJhoveNisoImageMetadataHelper.java

```diff
@@ -0,0 +1,523 @@
+        Map.entry("Sensor", NisoImageMetadata.SENSOR),
+        Map.entry("SourceXDimensionUnit", NisoImageMetadata.SOURCE_DIMENSION_UNIT),
+        Map.entry("SourceYDimensionUnit", NisoImageMetadata.SOURCE_DIMENSION_UNIT),
+        Map.entry("YCbCrPositioning", NisoImageMetadata.YCBCR_POSITIONING),
+        Map.entry("TargetType", NisoImageMetadata.TARGET_TYPE)
+    );
+
+    /**
+     * Getter functions that return an index into an indexed label array.
+     */
+    public static final Map<String, Function<NisoImageMetadata, Integer>> INDEXED_LABEL_PROPERTIES_GETTER_MAP = Map.ofEntries(
+        Map.entry("ColorSpace", (metadata) -> metadata.getColorSpace()),
+        Map.entry("CompressionScheme", (metadata) -> metadata.getCompressionScheme()),
+        Map.entry("SceneIlluminant", (metadata) -> metadata.getSceneIlluminant())
+    );
+
+    /**
+     * Label arrays for getter functions that return an index into an indexed label array.
+     */
+    public static final Map<String, String[]> INDEXED_LABEL_PROPERTIES_LABEL_ARRAY_MAP = Map.ofEntries(
+        Map.entry("ColorSpace", NisoImageMetadata.COLORSPACE),
+        Map.entry("CompressionScheme", NisoImageMetadata.COMPRESSION_SCHEME),
+        Map.entry("SceneIlluminant", NisoImageMetadata.SCENE_ILLUMINANT)
+    );
+
+    /**
+     * Index arrays for getter functions that return an index into an indexed label array.
+     */
+    public static final Map<String, int[]> INDEXED_LABEL_PROPERTIES_INDEX_ARRAY_MAP = Map.ofEntries(
+        Map.entry("ColorSpace", NisoImageMetadata.COLORSPACE_INDEX),
+        Map.entry("CompressionScheme", NisoImageMetadata.COMPRESSION_SCHEME_INDEX),
+        Map.entry("SceneIlluminant", NisoImageMetadata.SCENE_ILLUMINANT_INDEX)
+    );
+
+    /**
+     * Resolves a label by following the index through an index and label array.
+     * 
+     * @param value the index
+     * @param labelArray the label array
+     * @param indexArray the index array
+     * @return the label
+     */
+    private static Optional<String> parseLabelFromInt(int value, String[] labelArray, int[] indexArray) {
+        if (Objects.nonNull(value) && value != NisoImageMetadata.NULL) {
+            int index = ArrayUtils.indexOf(indexArray, value);
+            if (index != ArrayUtils.INDEX_NOT_FOUND) {
+                return Optional.of(labelArray[index]);
+            }
+            return Optional.of(String.valueOf(value));
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Extract indexed label properties and add them to the metadata map.
+     * 
+     * @param metadata the niso image matadata 
+     * @param map the simple string map that is being filled
+     */
+    private static void extractIndexedLabelProperties(NisoImageMetadata metadata, Map<String, String> map) {
+        // indexed label properties
+        for (Map.Entry<String, Function<NisoImageMetadata, Integer>> entry : INDEXED_LABEL_PROPERTIES_GETTER_MAP.entrySet()) {
```

Comment:
@solth, that's a case where I'd prefer `var`. But what do you think?
```suggestion
        for (var entry : INDEXED_LABEL_PROPERTIES_GETTER_MAP.entrySet()) {
```